src/main/java/net/sf/jabref/JabRef.java

```diff
@@ -97,6 +100,29 @@ public void start(String[] args) {
             System.setProperty("proxySet", "true");
         }
 
+        if (prefs.getBoolean(JabRefPreferences.USE_PROXY)
+                && prefs.getBoolean(JabRefPreferences.USE_PROXY_AUTHENTICATION)) {
+            Authenticator.setDefault(new Authenticator() {
+
+                @Override
+                protected PasswordAuthentication getPasswordAuthentication() {
+                    if (getRequestorType() == RequestorType.PROXY) {
+                        String prot = getRequestingProtocol().toLowerCase();
+                        String host = System.getProperty(prot + ".proxyHost", "");
+                        String port = System.getProperty(prot + ".proxyPort", "80");
+                        String user = System.getProperty(prot + ".proxyUser", "");
+                        String password = System.getProperty(prot + ".proxyPassword", "");
+                        if (getRequestingHost().equalsIgnoreCase(host)) {
+                            if (Integer.parseInt(port) == getRequestingPort()) {
+                                return new PasswordAuthentication(user, password.toCharArray());
+                            }
+                        }
+                    }
+                    return null;
+                }
+            });
+        }
```

Comment:
It would be perfect if you could extract these lines related to the proxy settings into a separate class (named `ProxyConfigurator` or something like that), as the JabRef class is already too big to handle well.